warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -45,19 +46,28 @@
     
     private static final Logger log = Logger.getLogger(GroupingTransform.class);
     
-    private BloomFilter<byte[]> bloom = null;
-    private Set<String> fields;
+    private final BloomFilter<byte[]> bloom;
+    private UniqueFields uniqueFields;
     private Multimap<String,String> modelMapping;
     
-    public UniqueTransform(Set<String> fields) {
-        this.fields = deconstruct(fields);
+    public UniqueTransform(UniqueFields uniqueFields) {
+        this.uniqueFields = uniqueFields;
+        uniqueFields.deconstructIdentifierFields();
```

Comment:
make this explicit ==> `this.uniqueFields.deconstructIdentifierFields();`

Reply:
Fixed.